controllers/propagator/replicatedpolicy_controller.go

```diff
@@ -0,0 +1,445 @@
+
+			return reconcile.Result{}, nil
+		}
+
+		log.Error(err, "Failed to get the root policy, requeueing")
+
+		return reconcile.Result{}, err
+	}
+
+	if rootPolicy.Spec.Disabled {
+		if replicatedExists {
+			if err := r.cleanUpReplicated(ctx, replicatedPolicy); err != nil {
+				if !k8serrors.IsNotFound(err) {
+					log.Error(err, "Failed to delete the disabled replicated policy, requeueing")
+
+					return reconcile.Result{}, err
+				}
+			}
+
+			log.Info("Disabled replicated policy deleted")
+
+			return reconcile.Result{}, nil
+		}
+
+		log.Info("Root policy is disabled, and replicated policy correctly not found.")
+
+		return reconcile.Result{}, nil
+	}
+
+	// calculate the decision for this specific cluster
+	decision, err := r.singleClusterDecision(ctx, rootPolicy, request.Namespace)
+	if err != nil {
+		log.Error(err, "Failed to determine if policy should be replicated, requeueing")
+
+		return reconcile.Result{}, err
+	}
+
+	// an empty decision means the policy should not be replicated
+	if decision.Cluster.ClusterName == "" {
+		if replicatedExists {
+			if err := r.cleanUpReplicated(ctx, replicatedPolicy); err != nil {
+				if !k8serrors.IsNotFound(err) {
+					log.Error(err, "Failed to remove the replicated policy for this managed cluster, requeueing")
+
+					return reconcile.Result{}, err
+				}
+			}
+
+			log.Info("Removed replicated policy from managed cluster")
+
+			return reconcile.Result{}, nil
+		}
+
+		log.Info("Replicated policy should not exist on this managed cluster, and does not.")
+
+		return reconcile.Result{}, nil
+	}
+
+	objsToWatch := getPolicySetDependencies(rootPolicy)
+
+	desiredReplicatedPolicy, err := r.buildReplicatedPolicy(rootPolicy, decision)
+	if err != nil {
+		log.Error(err, "Unable to build desired replicated policy, requeueing")
+
+		return reconcile.Result{}, err
+	}
+
+	// save the watcherError for later, so that the policy can still be updated now.
+	var watcherErr error
+
+	if policyHasTemplates(rootPolicy) {
+		if replicatedExists {
+			// If the replicated policy has an initialization vector specified, set it for processing
+			if initializationVector, ok := replicatedPolicy.Annotations[IVAnnotation]; ok {
+				tempAnnotations := desiredReplicatedPolicy.GetAnnotations()
+				if tempAnnotations == nil {
+					tempAnnotations = make(map[string]string)
+				}
+
+				tempAnnotations[IVAnnotation] = initializationVector
+
+				desiredReplicatedPolicy.SetAnnotations(tempAnnotations)
+			}
+		}
+
+		// resolve hubTemplate before replicating
+		// #nosec G104 -- any errors are logged and recorded in the processTemplates method,
+		// but the ignored status will be handled appropriately by the policy controllers on
+		// the managed cluster(s).
+		templObjsToWatch, _ := r.processTemplates(desiredReplicatedPolicy, decision.Cluster, rootPolicy)
+
+		for objID, val := range templObjsToWatch {
+			if val {
+				objsToWatch[objID] = true
+			}
+		}
+	}
```

Comment:
You should be able to use `templObjsToWatch` directly since it's a set. `val` would never be `false` based on the code in `processTemplates`.